assets/app/scripts/filters/resources.js

```diff
@@ -28,7 +38,25 @@ angular.module('openshiftConsole')
         }
       }
       if (resource && resource.metadata && resource.metadata.annotations) {
-        return resource.metadata.annotations[key];
+        // If the key's already in the annotation map, return it.
+        if (key in resource.metadata.annotations) {
+          return resource.metadata.annotations[key]
+        }
+        // If no mappings exist, and the key's not in the annotation map,
+        // return early.
+        var mappings = annotationMap[key];
```

Comment:
can simplify to `var mappings = annotationMap[key] || [];` and the loop will run 0 times